built-in-services/account/src/lib.rs

```diff
@@ -0,0 +1,261 @@
+            accounts.push(PayloadAccount {
+                address: item.address.clone(),
+                weight:  item.weight,
+            });
+        }
+
+        let response = GenerateAccountResponse {
+            accounts,
+            threshold: permission.threshold,
+            address: payload.user,
+        };
+
+        ServiceResponse::<GenerateAccountResponse>::from_succeed(response)
+    }
+
+    #[cycles(210_00)]
+    #[write]
+    fn generate_account(
+        &mut self,
+        ctx: ServiceContext,
+        payload: GenerateAccountPayload,
+    ) -> ServiceResponse<GenerateAccountResponse> {
+        if payload.accounts.is_empty() || payload.accounts.len() > MAX_PERMISSION_ACCOUNTS as usize
+        {
+            return ServiceResponse::<GenerateAccountResponse>::from_error(
+                110,
+                "accounts length must be [1,16]".to_owned(),
+            );
+        }
+
+        let mut weight_all = 0;
+        let mut accounts = Vec::<Account>::new();
+        for item in &payload.accounts {
+            weight_all += item.weight;
+            accounts.push(Account {
+                address:       item.address.clone(),
+                account_type:  ACCOUNT_TYPE_PUBLIC_KEY,
+                permission_id: 0,
+                weight:        item.weight,
+            });
+        }
+
+        if weight_all < payload.threshold || payload.threshold == 0 {
+            return ServiceResponse::<GenerateAccountResponse>::from_error(
+                110,
+                "accounts weight or threshold not valid".to_owned(),
+            );
+        }
+
+        let tx_hash = ctx.get_tx_hash().unwrap();
+
+        let keccak = HasherKeccak::new();
+        let addr_hash = Hash::from_bytes(Bytes::from(keccak.digest(&tx_hash.as_bytes())));
+        if addr_hash.is_err() {
+            return ServiceResponse::<GenerateAccountResponse>::from_error(
+                111,
+                "generate addr_hash from tx_hash failed".to_owned(),
+            );
+        }
+
+        let addr = Address::from_hash(addr_hash.unwrap());
+        if addr.is_err() {
+            return ServiceResponse::<GenerateAccountResponse>::from_error(
+                111,
+                "generate address from tx_hash failed".to_owned(),
+            );
+        }
+
+        let address = addr.unwrap();
+        let permission = Permission {
+            accounts,
+            threshold: payload.threshold,
+        };
+
+        self.sdk.set_account_value(&address, 0u8, permission);
+
+        let response = GenerateAccountResponse {
+            address,
+            accounts: payload.accounts,
+            threshold: payload.threshold,
+        };
+
+        ServiceResponse::<GenerateAccountResponse>::from_succeed(response)
+    }
+}
+
+fn verify_single_sig(tx_hash: &Hash, sig: &Hex, pubkey: &Hex) -> ServiceResponse<VerifyResponse> {
+    let data_hash = tx_hash.as_bytes();
+
+    let data_sig = hex::decode(sig.as_string_trim0x());
```

Comment:
The conversion is not necessary, try to use `tx_hash.as_bytes().to_vec()`

Reply:
ok